src/core/security/crypto/password-gen.ts

```diff
@@ -0,0 +1,34 @@
+export function generatePassword(
+  length: number = 12,
+  includeLowercase: boolean = true,
+  includeUppercase: boolean = true,
+  includeNumbers: boolean = true,
+  includeSymbols: boolean = true,
+) {
+  const lowercaseChars = 'abcdefghijklmnopqrstuvwxyz';
+  const uppercaseChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  const numberChars = '0123456789';
+  const symbolChars = '!@#$%^&*()_+-=';
+
+  let allowedChars = '';
+  let password = '';
+
+  allowedChars += includeLowercase ? lowercaseChars : '';
+  allowedChars += includeUppercase ? uppercaseChars : '';
+  allowedChars += includeNumbers ? numberChars : '';
+  allowedChars += includeSymbols ? symbolChars : '';
+
+  if (length <= 0) {
+    length = 12;
+  }
+  if (allowedChars.length === 0) {
+    allowedChars += lowercaseChars;
+  }
+
+  for (let i = 0; i < length; i++) {
+    const randomIndex = Math.floor(Math.random() * allowedChars.length);
+    password += allowedChars[randomIndex];
+  }
```

Comment:
poderia usar um `.reduce` aqui

Reply:
troquei o for por um reduce em um array vazio criado a partir de length (mais pelo aprendizado). Mas nesse caso não fica mais simples com o for mesmo ?